src/basic/process-util.c

```diff
@@ -1134,6 +1134,9 @@ static void restore_sigsetp(sigset_t **ssp) {
 
 int safe_fork_full(
                 const char *name,
+                int original_input_fd,
+                int original_output_fd,
+                int original_error_fd,
```

Comment:
hmm, here's an idea: make this `int original_stdio_fds[3]` instead. That way callers can just pass NULL here instead of the cumbersome `… -EBADF, -EBADF, -EBADF, …` if they want?

internally we'd treat NULL exactly like -EBADF, -EBADF, -EBADF is treated in your current code?